src/main/java/cn/nukkit/command/defaults/vanilla/GamemodeCommand.java

```diff
@@ -23,20 +29,18 @@ public GamemodeCommand(String name) {
                 "nukkit.command.gamemode.adventure;" +
                 "nukkit.command.gamemode.spectator;" +
                 "nukkit.command.gamemode.other");
-        this.commandParameters.clear();
-        this.commandParameters.add(new CommandParameter[]{
-                new CommandParameter("mode", CommandParamType.INT, false),
-                new CommandParameter("player", CommandParamType.TARGET, true)
-        });
-        this.commandParameters.add(new CommandParameter[]{
-                new CommandParameter("mode", new String[]{"survival", "s", "creative", "c",
-                        "adventure", "a", "spectator", "spc", "view", "v"}),
-                new CommandParameter("player", CommandParamType.TARGET, true)
-        });
+
+        registerOverload()
+                .then(requiredArg("mode", "GameMode", new String[]{"sp", "spectator", "a", "adventure", "c", "creative", "s", "survival"}))
```

Comment:
why not use Set<String> ?

Reply:
vararg parameter might be better (unless it cases conflicts)

Reply:
I plan to change this behaviour in the future anyway, so this is just a middle ground where it will work for now until i implement the more complicated behaviour.

Also Zed also mentioned about varargs - due to the name and enumName parameters, it looks too messy because to someone looking at the code there is no seperation between enum values and name. This doesnt matter though due to what i said above.

Reply:
And @NycuRO, in what world would i want to use a set?